nbconvert/filters/markdown_mistune.py

```diff
@@ -487,3 +489,39 @@ def render(self, source: str) -> str:
 def markdown2html_mistune(source: str) -> str:
     """Convert a markdown string to HTML using mistune"""
     return MarkdownWithMath(renderer=IPythonRenderer(escape=False)).render(source)
+
+
+class HeadingExtractor(MarkdownRenderer):
+    """A renderer to capture headings"""
+
+    def __init__(self):
+        """Initialize the class."""
+        super().__init__()
+        self.headings = []
+
+    def heading(self, text, level):
+        """Return an empty string for the headings to avoid outputting them."""
+        self.headings.append((level, text))
+        return ""
+
+
+def extract_titles_from_markdown_input(markdown_input):
+    """Create a Markdown parser with the HeadingExtractor renderer to collect all the headings of a notebook"""
+    """ The input argument is markdown_input that is a single string with all the markdown content concatenated """
+    """ The output is an array containing information about the headings such as their level, their text content, an identifier and a href that can be used in case of html converter.s"""
```

Comment:
```suggestion
    """Create a Markdown parser with the HeadingExtractor renderer to collect all the headings of a notebook
    The input argument is markdown_input that is a single string with all the markdown content concatenated
    The output is an array containing information about the headings such as their level, their text content, an identifier and a href that can be used in case of html converter.s"""
```